group.go

```diff
@@ -436,3 +449,36 @@ func (g *Group) Delete(recursive bool) error {
 	}
 	return nil
 }
+
+// AddMemberWithType adds a member to the Group providing its type
+// This method is recommended for performance when operating on remote groups
+func (g *Group) AddMemberWithType(uri, name string, isRelativeURI bool, objectType ObjectTypeEnum) error {
+	curi := C.CString(uri)
+	defer C.free(unsafe.Pointer(curi))
+
+	cname := C.CString(name)
+	defer C.free(unsafe.Pointer(cname))
+
+	var cRelative C.uint8_t
+	if isRelativeURI {
+		cRelative = 1
+	}
+
+	ret := C.tiledb_group_add_member_with_type(g.context.tiledbContext, g.group, curi, cRelative, cname, C.tiledb_object_t(objectType))
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Error adding member with type to group: %s", g.context.LastError())
+	}
+	return nil
+}
+
+// IsOpen returns true if the Group is open or false if the group is closed
+func (g *Group) IsOpen() (bool, error) {
+	var isOpen C.int32_t
+
+	ret := C.tiledb_group_is_open(g.context.tiledbContext, g.group, &isOpen)
+	if ret != C.TILEDB_OK {
+		return false, fmt.Errorf("Error checking if group is open: %s", g.context.LastError())
+	}
+
+	return isOpen > 0, nil
+}
```

Comment:
I would either return false or panic here in case of an error; I think it's excessive to return `(T, error)` for every single API regardless of how [trivial](https://github.com/TileDB-Inc/TileDB/blob/c0458631faeeeb31f83ff4ec1ef8b3f94606c6c2/tiledb/api/c_api/group/group_api.cc#L390-L398) it is, and has made using the API difficult. Several other similar APIs come to my mind and we should consider changing them in the future.

Reply:
Done